samples/nrf_desktop/src/modules/usb_state.c

```diff
@@ -28,8 +28,31 @@ static enum usb_state state;
 static u8_t hid_protocol = HID_PROTOCOL_REPORT;
 static struct device *usb_dev;
 
+static atomic_t feature_delivered;
+
 static int get_report(struct usb_setup_packet *setup, s32_t *len, u8_t **data)
 {
+	if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE)) {
+		u8_t request_value[2];
+		sys_put_le16(setup->wValue, request_value);
+
+		if (request_value[1] == 0x03) {
+			/* Request for feature report */
+			if (request_value[0] == REPORT_ID_USER_CONFIG) {
+				static u8_t ready;
+
+				ready = atomic_get(&feature_delivered);
+				*len = sizeof(ready);
+				*data = (u8_t *)(&ready);
```

Comment:
are you sure this function won't be called again before async user of `ready` is done with it?

Reply:
This function will be called again when python tool polls on USB - and it will get info that BLE is not ready with the transmission yet (so tool will wait for next poll to send next config).